src/hnsw.c

```diff
@@ -94,6 +96,29 @@ hnswcostestimate(PlannerInfo *root, IndexPath *path, double loop_count,
 		return;
 	}
 
+	/* Get the selectivity of non-index conditions */
+	foreach(lc, path->indexinfo->indrestrictinfo)
+	{
+		RestrictInfo *rinfo = lfirst(lc);
+
+		if (rinfo->norm_selec >= 0 && rinfo->norm_selec <= 1 && rinfo->norm_selec != (Selectivity) DEFAULT_INEQ_SEL)
+			selectivity *= rinfo->norm_selec;
+	}
+
+	/*
+	 * Do not use index if limit + offset > expected tuples unless
+	 * enable_seqscan = off
+	 */
+	if (root->limit_tuples > hnsw_ef_search * selectivity)
+	{
+		*indexStartupCost = 1.0e10 - 1;
+		*indexTotalCost = 1.0e10 - 1;
```

Comment:
Suggestion: a comment that this value is derived from `disable_cost` in `src/backend/optimizer/path/costsize.c`,  should that value ever change. Perhaps also move it up into a const at the top of this file.

Reply:
Sounds good - will add more comments / make const once we decide on an approach.